oneclick.py

```diff
@@ -589,6 +682,7 @@ def isbn_info_to_metadata(cls, book, include_bibliographic=True, include_formats
                     subjects.append(subject)
 
             # audience options are: adult, beginning-reader, childrens, young-adult
+            # NOTE: audience can be set to "Adult" while publisher is "HarperTeen".
```

Comment:
I don't think this is a NOTE situation. Either it's a bug in core/classifier.py (in which case you should file the bug) or there's some other aspect of the metadata that outweighs the "HarperTeen" -> "Young Adult" mapping in classifier.py.

My guess is that OneClick is sending HarperTeen as the publisher, but classifier.py is expecting Harper Collins as the publisher and HarperTeen as the imprint. If this is true, the best solution would be to change `weigh_metadata` in classifier.py to collapse the distinction between 'publisher' and 'imprint' -- always check both fields, since we can't be sure which one might say "HarperTeen".